--- ncurses-5.9/Ada95/aclocal.m4.orig	2011-04-01 08:32:36.000000000 +0900
+++ ncurses-5.9/Ada95/aclocal.m4	2013-01-20 12:23:03.145326427 +0900
@@ -66,10 +66,10 @@
 dnl CF_ADD_ADAFLAGS version: 1 updated: 2010/06/19 15:22:18
 dnl ---------------
 dnl Add to $ADAFLAGS, which is substituted into makefile and scripts.
-AC_DEFUN([CF_ADD_ADAFLAGS],[
- 	ADAFLAGS="$ADAFLAGS $1"
-	AC_SUBST(ADAFLAGS)
-])dnl
+dnl AC_DEFUN([CF_ADD_ADAFLAGS],[
+dnl  	ADAFLAGS="$ADAFLAGS $1"
+dnl 	AC_SUBST(ADAFLAGS)
+dnl ])dnl
 dnl ---------------------------------------------------------------------------
 dnl CF_ADD_CFLAGS version: 10 updated: 2010/05/26 05:38:42
 dnl -------------
