diff -ru gtk+-2.24.22.orig/gtk/Makefile.am gtk+-2.24.22/gtk/Makefile.am
--- gtk+-2.24.22.orig/gtk/Makefile.am	2013-10-05 07:02:29.000000000 +0900
+++ gtk+-2.24.22/gtk/Makefile.am	2013-12-25 00:39:10.741377478 +0900
@@ -1040,7 +1040,7 @@
 #
 bin_PROGRAMS = \
 	gtk-query-immodules-2.0 \
-	gtk-update-icon-cache
+	gtk-update-icon-cache$(EXEEXT)
 
 bin_SCRIPTS = gtk-builder-convert
 
diff -ru gtk+-2.24.22.orig/modules/input/Makefile.am gtk+-2.24.22/modules/input/Makefile.am
--- gtk+-2.24.22.orig/modules/input/Makefile.am	2013-10-05 07:02:29.000000000 +0900
+++ gtk+-2.24.22/modules/input/Makefile.am	2013-12-25 00:40:00.596200387 +0900
@@ -202,8 +202,8 @@
 	@if $(RUN_QUERY_IMMODULES_TEST) ; then \
 	  echo $(mkinstalldirs) $(DESTDIR)$(sysconfdir)/gtk-2.0 ; \
 	  $(mkinstalldirs) $(DESTDIR)$(sysconfdir)/gtk-2.0 ; \
-	  echo "$(DESTDOR)$(bindir)/gtk-query-immodules-2.0 > $(DESTDIR)$(libdir)/gtk-2.0/$(GTK_BINARY_VERSION)/immodules.cache" ; \
-	  $(DESTDIR)/$(bindir)/gtk-query-immodules-2.0 > $(DESTDIR)$(libdir)/gtk-2.0/$(GTK_BINARY_VERSION)/immodules.cache ; \
+	  echo "$(DESTDOR)$(bindir)/gtk-query-immodules-2.0$(EXEEXT) > $(DESTDIR)$(libdir)/gtk-2.0/$(GTK_BINARY_VERSION)/immodules.cache" ; \
+	  $(DESTDIR)/$(bindir)/gtk-query-immodules-2.0$(EXEEXT) > $(DESTDIR)$(libdir)/gtk-2.0/$(GTK_BINARY_VERSION)/immodules.cache ; \
 	else \
 	  echo "***" ; \
 	  echo "*** Warning: immodules.cache not built" ; \
@@ -253,7 +253,7 @@
 included-modules: $(noinst_LTLIBRARIES)
 
 immodules.cache: Makefile.am $(module_LTLIBRARIES)
-	$(top_builddir)/gtk/gtk-query-immodules-2.0 $(module_LTLIBRARIES) > immodules.cache
+	$(top_builddir)/gtk/gtk-query-immodules-2.0$(EXEEXT) $(module_LTLIBRARIES) > immodules.cache
 
 CLEANFILES = immodules.cache
 
