diff -ru glib-2.38.2.orig/gio/tests/Makefile.am glib-2.38.2/gio/tests/Makefile.am
--- glib-2.38.2.orig/gio/tests/Makefile.am	2013-11-12 14:30:22.000000000 +0900
+++ glib-2.38.2/gio/tests/Makefile.am	2013-12-23 15:13:59.328468099 +0900
@@ -431,7 +431,7 @@
 if CROSS_COMPILING
   glib_compile_resources=$(GLIB_COMPILE_RESOURCES)
 else
-  glib_compile_resources=$(top_builddir)/gio/glib-compile-resources
+  glib_compile_resources=$(top_builddir)/gio/glib-compile-resources$(EXEEXT)
 endif
 
 resources.o: test_resources2.h
diff -ru glib-2.38.2.orig/tests/gobject/Makefile.am glib-2.38.2/tests/gobject/Makefile.am
--- glib-2.38.2.orig/tests/gobject/Makefile.am	2013-11-08 00:29:13.000000000 +0900
+++ glib-2.38.2/tests/gobject/Makefile.am	2013-12-23 15:12:21.190876890 +0900
@@ -51,7 +51,7 @@
 if CROSS_COMPILING
   glib_genmarshal=$(GLIB_GENMARSHAL)
 else
-  glib_genmarshal=$(top_builddir)/gobject/glib-genmarshal
+  glib_genmarshal=$(top_builddir)/gobject/glib-genmarshal$(EXEEXT)
 endif
 
 testmarshal.h: stamp-testmarshal.h
