diff -ru glib-2.48.0.orig/gio/tests/Makefile.am glib-2.48.0/gio/tests/Makefile.am
--- glib-2.48.0.orig/gio/tests/Makefile.am	2016-03-23 00:15:18.000000000 +0900
+++ glib-2.48.0/gio/tests/Makefile.am	2016-04-01 23:57:17.701727371 +0900
@@ -532,7 +532,7 @@
 libresourceplugin_la_LDFLAGS += -rpath /
 endif
 
-glib_compile_resources=$(top_builddir)/gio/glib-compile-resources
+glib_compile_resources=$(top_builddir)/gio/glib-compile-resources$(EXEEXT)
 
 resources.o: test_resources2.h
 test_resources.c: test2.gresource.xml Makefile $(shell $(glib_compile_resources) --sourcedir=$(srcdir) --generate-dependencies $(srcdir)/test2.gresource.xml)
diff -ru glib-2.48.0.orig/gobject/tests/Makefile.am glib-2.48.0/gobject/tests/Makefile.am
--- glib-2.48.0.orig/gobject/tests/Makefile.am	2016-02-29 23:32:08.000000000 +0900
+++ glib-2.48.0/gobject/tests/Makefile.am	2016-04-02 00:03:17.344944211 +0900
@@ -36,7 +36,7 @@
 # cross-compiling
 
 if !CROSS_COMPILING
-glib_genmarshal=$(top_builddir)/gobject/glib-genmarshal
+glib_genmarshal=$(top_builddir)/gobject/glib-genmarshal$(EXEEXT)
 
 test_programs += signals
 signals_SOURCES = signals.c
diff -ru glib-2.48.0.orig/tests/gobject/Makefile.am glib-2.48.0/tests/gobject/Makefile.am
--- glib-2.48.0.orig/tests/gobject/Makefile.am	2016-02-24 07:25:37.000000000 +0900
+++ glib-2.48.0/tests/gobject/Makefile.am	2016-04-02 00:00:07.259367148 +0900
@@ -51,7 +51,7 @@
 # The marshal test requires running a binary, which means we cannot
 # build it when cross-compiling
 if !CROSS_COMPILING
-glib_genmarshal=$(top_builddir)/gobject/glib-genmarshal
+glib_genmarshal=$(top_builddir)/gobject/glib-genmarshal$(EXEEXT)
 
 testmarshal.h: stamp-testmarshal.h
 	@true
