--- glib-networking-2.28.7/configure.ac.orig	2011-05-24 04:41:01.000000000 +0900
+++ glib-networking-2.28.7/configure.ac	2012-07-22 09:40:31.530793704 +0900
@@ -66,7 +66,7 @@
 dnl **************************************
 AC_ARG_WITH(gnome-proxy,
     [AC_HELP_STRING([--with-gnome-proxy],
-                    [support for GNOME proxy configuration @<:@default=check@:>@])]
+                    [support for GNOME proxy configuration @<:@default=check@:>@])],
     [],
     [with_gnome_proxy=check])
 AS_IF([test "x$with_gnome_proxy" != "xno"],
--- /dev/null	2012-07-22 07:44:28.402150762 +0900
+++ glib-networking-2.28.7/m4/.keepme	2012-07-22 09:43:43.993712447 +0900
@@ -0,0 +1 @@
+keepme
