diff --git a/gio/tests/Makefile.am b/gio/tests/Makefile.am
index 1f0bed7b6..6fa002685 100644
--- a/gio/tests/Makefile.am
+++ b/gio/tests/Makefile.am
@@ -578,7 +578,7 @@ if !ENABLE_INSTALLED_TESTS
 libresourceplugin_la_LDFLAGS += -rpath /
 endif
 
-glib_compile_resources=$(top_builddir)/gio/glib-compile-resources
+glib_compile_resources=$(top_builddir)/gio/glib-compile-resources$(EXEEXT)
 
 test-generated.txt: test1.txt
 	$(AM_V_GEN) echo "Generated" > $@ && \
