--- gnutls-3.2.7/gl/string.in.h.orig	2013-11-11 02:59:13.000000000 +0900
+++ gnutls-3.2.7/gl/string.in.h	2013-12-21 14:39:45.954008000 +0900
@@ -687,7 +687,7 @@
 #   undef strtok_r
 #  endif
 #  if ! @HAVE_DECL_STRTOK_R@
-_GL_FUNCDECL_SYS (strtok_r, char *,
+_GL_FUNCDECL_SYS ((strtok_r), char *,
                   (char *restrict s, char const *restrict delim,
                    char **restrict save_ptr)
                   _GL_ARG_NONNULL ((2, 3)));
