--- gtk+-3.6.4/gtk/Makefile.am.orig	2013-01-07 19:50:02.000000000 +0900
+++ gtk+-3.6.4/gtk/Makefile.am	2013-01-19 21:55:56.629173491 +0900
@@ -1644,7 +1644,7 @@
 if USE_EXTERNAL_ICON_CACHE
 gtk_update_icon_cache_program = $(GTK_UPDATE_ICON_CACHE)
 else
-gtk_update_icon_cache_program = ./gtk-update-icon-cache
+gtk_update_icon_cache_program = ./gtk-update-icon-cache$(EXEEXT)
 endif
 
 gtkbuiltincache.h: @REBUILD@ stamp-icons
--- gtk+-3.6.4/modules/input/Makefile.am.orig	2013-01-19 22:20:06.768724437 +0900
+++ gtk+-3.6.4/modules/input/Makefile.am	2013-01-19 22:20:25.868239314 +0900
@@ -186,8 +186,8 @@
 	$(AM_V_GEN) if $(RUN_QUERY_IMMODULES_TEST) ; then \
 	  echo $(mkinstalldirs) $(DESTDIR)$(libdir)/gtk-3.0/3.0.0 ; \
 	  $(mkinstalldirs) $(DESTDIR)$(libdir)/gtk-3.0/3.0.0 ; \
-	  echo "$(top_builddir)/gtk/gtk-query-immodules-3.0 > $(DESTDIR)$(libdir)/gtk-3.0/3.0.0/immodules.cache" ; \
-	  $(top_builddir)/gtk/gtk-query-immodules-3.0 > $(DESTDIR)$(libdir)/gtk-3.0/3.0.0/immodules.cache ; \
+	  echo "$(top_builddir)/gtk/gtk-query-immodules-3.0$(EXEEXT) > $(DESTDIR)$(libdir)/gtk-3.0/3.0.0/immodules.cache" ; \
+	  $(top_builddir)/gtk/gtk-query-immodules-3.0$(EXEEXT) > $(DESTDIR)$(libdir)/gtk-3.0/3.0.0/immodules.cache ; \
 	else \
 	  echo "***" ; \
 	  echo "*** Warning: immodules.cache not built" ; \
@@ -235,7 +235,7 @@
 included-modules: $(noinst_LTLIBRARIES)
 
 immodules.cache: Makefile.am $(module_LTLIBRARIES)
-	$(AM_V_GEN) $(top_builddir)/gtk/gtk-query-immodules-3.0 $(module_LTLIBRARIES) > immodules.cache
+	$(AM_V_GEN) $(top_builddir)/gtk/gtk-query-immodules-3.0$(EXEEXT) $(module_LTLIBRARIES) > immodules.cache
 
 EXTRA_DIST += README.multipress
 
