diff -ru gtk+-3.16.2.orig/demos/gtk-demo/Makefile.am gtk+-3.16.2/demos/gtk-demo/Makefile.am
--- gtk+-3.16.2.orig/demos/gtk-demo/Makefile.am	2015-04-11 02:01:31.000000000 +0900
+++ gtk+-3.16.2/demos/gtk-demo/Makefile.am	2015-09-06 17:09:10.900427077 +0900
@@ -173,7 +173,7 @@
 dist_appsicon48_DATA = data/48x48/gtk3-demo.png data/48x48/gtk3-demo-symbolic.symbolic.png
 dist_appsicon256_DATA = data/256x256/gtk3-demo.png data/256x256/gtk3-demo-symbolic.symbolic.png
 
-update_icon_cache = gtk-update-icon-cache --ignore-theme-index --force
+update_icon_cache = gtk-update-icon-cache$(EXEEXT) --ignore-theme-index --force
 
 install-data-hook: install-update-icon-cache
 uninstall-hook: uninstall-update-icon-cache
diff -ru gtk+-3.16.2.orig/demos/widget-factory/Makefile.am gtk+-3.16.2/demos/widget-factory/Makefile.am
--- gtk+-3.16.2.orig/demos/widget-factory/Makefile.am	2015-04-11 02:01:31.000000000 +0900
+++ gtk+-3.16.2/demos/widget-factory/Makefile.am	2015-09-06 17:12:11.164020089 +0900
@@ -47,7 +47,7 @@
 dist_appsicon48_DATA = data/48x48/gtk3-widget-factory.png data/48x48/gtk3-widget-factory-symbolic.symbolic.png
 dist_appsicon256_DATA = data/256x256/gtk3-widget-factory.png data/256x256/gtk3-widget-factory-symbolic.symbolic.png
 
-update_icon_cache = gtk-update-icon-cache --ignore-theme-index --force
+update_icon_cache = gtk-update-icon-cache$(EXEEXT) --ignore-theme-index --force
 
 install-data-hook: install-update-icon-cache
 uninstall-hook: uninstall-update-icon-cache
