diff --git a/giscanner/ccompiler.py b/giscanner/ccompiler.py
index 57df9d6..099001a 100644
--- a/giscanner/ccompiler.py
+++ b/giscanner/ccompiler.py
@@ -293,6 +293,7 @@ class CCompiler(object):
             for line in o.decode('ascii').splitlines():
                 if line.startswith('libraries: '):
                     libsearch = line[len('libraries: '):].split(os.pathsep)
+        libsearch = options.library_paths + libsearch
 
         shlibs = []
         not_resolved = []
