diff -ru gst-plugins-bad-1.8.0.orig/gst-libs/gst/gl/Makefile.am gst-plugins-bad-1.8.0/gst-libs/gst/gl/Makefile.am
--- gst-plugins-bad-1.8.0.orig/gst-libs/gst/gl/Makefile.am	2016-03-12 19:41:42.000000000 +0900
+++ gst-plugins-bad-1.8.0/gst-libs/gst/gl/Makefile.am	2016-04-03 15:59:50.446978896 +0900
@@ -160,7 +160,7 @@
 		--c-include "gst/gl/gl.h" \
 		-I$(top_srcdir)/gst-libs \
 		-I$(top_builddir)/gst-libs \
-		$(GST_PLUGINS_BASE_CFLAGS) \
+		`PKG_CONFIG_PATH="$(GST_PKG_CONFIG_PATH)" $(PKG_CONFIG) --cflags-only-I gstreamer-base-@GST_API_VERSION@` \
 		$(GL_CFLAGS) \
 		--add-include-path=`PKG_CONFIG_PATH="$(GST_PKG_CONFIG_PATH)" $(PKG_CONFIG) --variable=girdir gstreamer-@GST_API_VERSION@` \
 		--add-include-path=`PKG_CONFIG_PATH="$(GST_PKG_CONFIG_PATH)" $(PKG_CONFIG) --variable=girdir gstreamer-base-@GST_API_VERSION@` \
