diff -ru gst-plugins-bad-1.8.0.orig/sys/directsound/Makefile.am gst-plugins-bad-1.8.0/sys/directsound/Makefile.am
--- gst-plugins-bad-1.8.0.orig/sys/directsound/Makefile.am	2015-04-13 15:56:11.000000000 +0900
+++ gst-plugins-bad-1.8.0/sys/directsound/Makefile.am	2016-04-03 17:42:06.389387506 +0900
@@ -5,7 +5,7 @@
 	$(GST_PLUGINS_BASE_CFLAGS) $(DIRECTX_CFLAGS)
 libgstdirectsoundsrc_la_LIBADD = \
 	$(GST_BASE_LIBS) $(GST_PLUGINS_BASE_LIBS) -lgstaudio-$(GST_API_VERSION) \
-	$(DIRECTX_LDFLAGS) $(DIRECTSOUND_LIBS)
+	$(DIRECTX_LDFLAGS) $(DIRECTSOUND_LIBS) -lole32
 libgstdirectsoundsrc_la_LDFLAGS = $(GST_PLUGIN_LDFLAGS)
 libgstdirectsoundsrc_la_LIBTOOLFLAGS = $(GST_PLUGIN_LIBTOOLFLAGS)
 
