diff -ruN libmad-0.15.1b.orig/AUTHORS libmad-0.15.1b/AUTHORS
--- libmad-0.15.1b.orig/AUTHORS	1970-01-01 09:00:00.000000000 +0900
+++ libmad-0.15.1b/AUTHORS	2012-01-07 19:35:23.810029179 +0900
@@ -0,0 +1 @@
+for autoconf.
diff -ruN libmad-0.15.1b.orig/ChangeLog libmad-0.15.1b/ChangeLog
--- libmad-0.15.1b.orig/ChangeLog	1970-01-01 09:00:00.000000000 +0900
+++ libmad-0.15.1b/ChangeLog	2012-01-07 19:36:28.244351603 +0900
@@ -0,0 +1 @@
+for autoconf.
diff -ruN libmad-0.15.1b.orig/Makefile.am libmad-0.15.1b/Makefile.am
--- libmad-0.15.1b.orig/Makefile.am	2004-02-17 11:02:03.000000000 +0900
+++ libmad-0.15.1b/Makefile.am	2012-01-07 19:30:38.245466821 +0900
@@ -101,7 +101,7 @@
 
 version_info =		$(version_current):$(version_revision):$(version_age)
 
-libmad_la_LDFLAGS =	-version-info $(version_info)
+libmad_la_LDFLAGS =	-version-info $(version_info) -no-undefined
 
 mad.h: config.status config.h Makefile.am  \
 		$(srcdir)/COPYRIGHT $(srcdir)/mad.h.sed $(exported_headers)
diff -ruN libmad-0.15.1b.orig/NEWS libmad-0.15.1b/NEWS
--- libmad-0.15.1b.orig/NEWS	1970-01-01 09:00:00.000000000 +0900
+++ libmad-0.15.1b/NEWS	2012-01-07 19:35:28.429908891 +0900
@@ -0,0 +1 @@
+for autoconf.
