--- flac-1.2.1/src/libFLAC/Makefile.am.orig	2007-09-15 06:06:17.000000000 +0900
+++ flac-1.2.1/src/libFLAC/Makefile.am	2012-01-07 14:15:45.582310444 +0900
@@ -96,7 +96,7 @@
 	ogg_mapping.c
 endif
 # see 'http://www.gnu.org/software/libtool/manual.html#Libtool-versioning' for numbering convention
-libFLAC_la_LDFLAGS = -version-info 10:0:2 -lm $(LOCAL_EXTRA_LDFLAGS)
+libFLAC_la_LDFLAGS = -version-info 10:0:2 -lws2_32 $(LOCAL_EXTRA_LDFLAGS) -no-undefined
 libFLAC_la_SOURCES = \
 	bitmath.c \
 	bitreader.c \
