diff -ru gtk+-3.10.6.orig/gdk/Makefile.am gtk+-3.10.6/gdk/Makefile.am
--- gtk+-3.10.6.orig/gdk/Makefile.am	2013-12-05 12:17:39.000000000 +0900
+++ gtk+-3.10.6/gdk/Makefile.am	2013-12-25 23:10:14.691695701 +0900
@@ -179,8 +179,8 @@
 if USE_WIN32
 libgdk_3_la_SOURCES += gdkkeynames.c
 libgdk_3_la_LIBADD += win32/libgdk-win32.la
-libgdk_3_la_DEPENDENCIES = win32/libgdk-win32.la win32/rc/gdk-win32-res.o gdk.def
-libgdk_3_la_LDFLAGS += -Wl,win32/rc/gdk-win32-res.o -export-symbols $(srcdir)/gdk.def
+libgdk_3_la_DEPENDENCIES = win32/libgdk-win32.la win32/rc/gdk-win32-res.o
+libgdk_3_la_LDFLAGS += -Wl,win32/rc/gdk-win32-res.o -no-undefined
 endif # USE_WIN32
 
 if USE_BROADWAY
@@ -270,17 +270,6 @@
 
 endif # HAVE_INTROSPECTION
 
-if OS_WIN32
-install-def-file: gdk.def
-	mkdir -p $(DESTDIR)$(libdir)
-	$(INSTALL) $(srcdir)/gdk.def $(DESTDIR)$(libdir)/gdk-win32-3.0.def
-uninstall-def-file:
-	-rm $(DESTDIR)$(libdir)/gdk-win32-3.0.def
-else
-install-def-file:
-uninstall-def-file:
-endif
-
 if MS_LIB_AVAILABLE
 noinst_DATA = gdk-win32-$(GTK_API_VERSION).lib
 
@@ -382,9 +371,9 @@
 
 DISTCLEANFILES = gdkconfig.h stamp-gc-h
 
-install-data-local: install-ms-lib install-def-file
+install-data-local: install-ms-lib
 
-uninstall-local: uninstall-ms-lib uninstall-def-file
+uninstall-local: uninstall-ms-lib
 	rm -f $(DESTDIR)$(configexecincludedir)/gdkconfig.h
 
 # if srcdir!=builddir, clean out maintainer-clean files from builddir
diff -ru gtk+-3.10.6.orig/gtk/Makefile.am gtk+-3.10.6/gtk/Makefile.am
--- gtk+-3.10.6.orig/gtk/Makefile.am	2013-12-05 12:51:06.000000000 +0900
+++ gtk+-3.10.6/gtk/Makefile.am	2013-12-25 23:10:26.047429593 +0900
@@ -61,29 +61,18 @@
 endif
 
 if OS_WIN32
-gtk_def = gtk.def
-gtk_win32_symbols = -export-symbols $(srcdir)/gtk.def
-
 gtk_win32_res = gtk-win32-res.o
 gtk_win32_res_ldflag = -Wl,gtk-win32-res.o
 
 gtk-win32-res.o : gtk-win32.rc
 	$(WINDRES) gtk-win32.rc $@
-
-install-def-file: gtk.def
-	$(INSTALL) $(srcdir)/gtk.def $(DESTDIR)$(libdir)/gtk-win32-3.0.def
-uninstall-def-file:
-	-rm $(DESTDIR)$(libdir)/gtk-win32-3.0.def
-else
-install-def-file:
-uninstall-def-file:
 endif
 
 if MS_LIB_AVAILABLE
 noinst_DATA = gtk-win32-$(GTK_API_VERSION).lib
 
-gtk-win32-$(GTK_API_VERSION).lib: libgtk-win32-$(GTK_API_VERSION).la gtk.def
-	lib -machine:@LIB_EXE_MACHINE_FLAG@ -name:libgtk-win32-$(GTK_API_VERSION)-@LT_CURRENT_MINUS_AGE@.dll -def:gtk.def -out:$@
+gtk-win32-$(GTK_API_VERSION).lib: libgtk-win32-$(GTK_API_VERSION).la
+	lib -machine:@LIB_EXE_MACHINE_FLAG@ -name:libgtk-win32-$(GTK_API_VERSION)-@LT_CURRENT_MINUS_AGE@.dll -out:$@
 
 install-ms-lib:
 	$(INSTALL) gtk-win32-$(GTK_API_VERSION).lib $(DESTDIR)$(libdir)
@@ -1233,8 +1222,8 @@
 if USE_WIN32
 libgtk_3_la_LIBADD += -lole32 -lgdi32 -lcomdlg32 -lwinspool -lcomctl32
 libgtk_3_la_LDFLAGS += -Wl,-luuid
-libgtk_3_la_DEPENDENCIES += $(gtk_def) $(gtk_win32_res) $(deps)
-libgtk_target_ldflags = $(gtk_win32_res_ldflag) $(gtk_win32_symbols)
+libgtk_3_la_DEPENDENCIES += $(gtk_win32_res) $(deps)
+libgtk_target_ldflags = $(gtk_win32_res_ldflag)
 endif
 
 install-exec-hook:
@@ -1286,13 +1275,13 @@
 	rm libgtk.vs10.sourcefiles.filters
 
 # Install a RC file for the default GTK+ theme, and key themes
-install-data-local: install-ms-lib install-def-file install-mac-key-theme
+install-data-local: install-ms-lib install-mac-key-theme
 	$(MKDIR_P) $(DESTDIR)$(datadir)/themes/Default/gtk-3.0
 	$(INSTALL_DATA) $(srcdir)/gtk-keys.css.default $(DESTDIR)$(datadir)/themes/Default/gtk-3.0/gtk-keys.css
 	$(MKDIR_P) $(DESTDIR)$(datadir)/themes/Emacs/gtk-3.0
 	$(INSTALL_DATA) $(srcdir)/gtk-keys.css.emacs $(DESTDIR)$(datadir)/themes/Emacs/gtk-3.0/gtk-keys.css
 
-uninstall-local: uninstall-ms-lib uninstall-def-file uninstall-mac-key-theme
+uninstall-local: uninstall-ms-lib uninstall-mac-key-theme
 	rm -f $(DESTDIR)$(datadir)/themes/Raleigh/gtk-3.0/gtk.css
 	rm -f $(DESTDIR)$(datadir)/themes/Default/gtk-3.0/gtk-keys.css
 	rm -f $(DESTDIR)$(datadir)/themes/Emacs/gtk-3.0/gtk-keys.css
