diff -ru harfbuzz-0.9.17.orig/configure.ac harfbuzz-0.9.17/configure.ac
--- harfbuzz-0.9.17.orig/configure.ac	2013-05-20 22:21:57.000000000 +0900
+++ harfbuzz-0.9.17/configure.ac	2013-05-26 09:01:41.881876729 +0900
@@ -248,7 +248,7 @@
 LIBS="$LIBS -lusp10 -lgdi32"
 AC_LINK_IFELSE([AC_LANG_PROGRAM(
 	[[
-	 #define _WIN32_WINNT 0x0600
+	 #define _WIN32_WINNT 0x0501
 	 #include <windows.h>
 	 #include <usp10.h>
 	 ]],
