diff -ru pango-1.34.0.orig/pango/pangocairo.def pango-1.34.0/pango/pangocairo.def
--- pango-1.34.0.orig/pango/pangocairo.def	2013-01-24 12:39:14.000000000 +0900
+++ pango-1.34.0/pango/pangocairo.def	2013-05-18 20:28:04.017761164 +0900
@@ -8,8 +8,8 @@
 	pango_cairo_create_context
 	pango_cairo_create_layout
 	pango_cairo_error_underline_path
-;	pango_cairo_fc_font_get_type
-;	pango_cairo_fc_font_map_get_type
+	pango_cairo_fc_font_get_type
+	pango_cairo_fc_font_map_get_type
 	pango_cairo_font_get_type
 	pango_cairo_font_get_scaled_font
 	pango_cairo_font_map_create_context
